lib_comfyui/private/external_code.py

```diff
@@ -0,0 +1,203 @@
+import dataclasses
+from pathlib import Path
+from typing import List, Tuple, Union, Any, Optional
+from lib_comfyui import global_state
+
+
+ALL_TABS = ...
+Tabs = Union[str, Tuple[str]]
+
+
+@dataclasses.dataclass
+class WorkflowType:
+    """
+    Describes a unique type of ComfyUI workflow
+    """
+
+    base_id: str
+    display_name: str
+    tabs: Tabs = ('txt2img', 'img2img')
+    default_workflow: Union[str, Path] = "null"
+
+    def __post_init__(self):
+        if isinstance(self.tabs, str):
+            self.tabs = (self.tabs,)
+
+        assert self.tabs, "tabs must not be empty"
+
+        if isinstance(self.default_workflow, Path):
+            with open(str(self.default_workflow), 'r') as f:
+                self.default_workflow = f.read()
+
+    def get_ids(self, tabs: Tabs = ALL_TABS) -> List[str]:
```

Comment:
can you explain why each and every workflow type can have many ids?

Reply:
One id per tab. For example, the Postprocess workflow type has ids `postprocess_txt2img` and `postprocess_img2img`

Reply:
Both are the same workflow type, but they appear on different tabs and so they need different ids so that python can target a single one of them at a time.